File: Questao-3/Questao3.py
def dfs(a, grafo, visitados):
    visitados[a] = 1
    for j in range(E):
        if grafo[a][j]:
            if not visitados[j]:
                dfs(j, grafo, visitados)

E, L = [int(x) for x in input().split(' ')]
n = 1
while E != 0:

    grafo = [([0]*E).copy() for x in range(E)]
    
    while L:
        X, Y = [int(x) for x in input().split(' ')]
        grafo[X-1][Y-1] = 1
        L -= 1
    
    visitados = [0] * E
    dfs(0, grafo, visitados)
    print("Teste", n)
    print("normal\n" if sum(visitados) == E else "falha\n")
    n += 1
    E, L = [int(x) for x in input().split(' ')]

File: Questao-3/Questao3.c
#include <stdio.h>
#include <string.h>

int E, L, X, Y, falha, n = 1;

void dfs(int a, int grafo[E][E], int visitados[E]){
    visitados[a] = 1;
    for (int j = 0; j < E; j++){
        if (grafo[a][j]){
            if (!visitados[j]){
                dfs(j, grafo, visitados);
            }
        }
    }
}

int main(){
    scanf("%d %d", &E, &L);
    
    while(E != 0){
        int grafo[E][E];
        memset(grafo, 0, sizeof(grafo));

        while(L--){
            scanf("%d %d", &X, &Y);
            grafo[X-1][Y-1] = 1;
        }

        int visitados[E];
        memset(visitados, 0, sizeof(visitados));

        dfs(0, grafo, visitados);
        falha = 0;
        for(int i = 0; i < E; i++){
            if (!visitados[i]){
                falha = 1;
                break;
            }
        }
        if (falha) printf("Teste %d\nfalha\n\n", n);
        else printf("Teste %d\nnormal\n\n", n);
        n++;

        scanf("%d %d", &E, &L);
    }
    return 0;
}